Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -211,52 +472,88 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
         // force element is enabled.
         const Array_<bool>& forceEnabled = Value< Array_<bool> >::downcast
                                     (getDiscreteVariable(s, forceEnabledIndex));
+        const Array_<bool>& parallelEnabled = Value< Array_<bool> >::downcast
+                                    (getDiscreteVariable(s, parallelEnabledIndex));
 
         // Get access to System-global force cache arrays.
-        Vector_<SpatialVec>&   rigidBodyForces = 
+        Vector_<SpatialVec>&   rigidBodyForces =
                                     mbs.updRigidBodyForces(s, Stage::Dynamics);
-        Vector_<Vec3>&         particleForces  = 
+        Vector_<Vec3>&         particleForces  =
                                     mbs.updParticleForces (s, Stage::Dynamics);
-        Vector&                mobilityForces  = 
+        Vector&                mobilityForces  =
                                     mbs.updMobilityForces (s, Stage::Dynamics);
 
+        // Track the enabled forces and whether they should be parallelized.
```

Comment:
I originally cached the parallel and non-parallel forces at Stage::Instance but ran into tests failing. I looked into the exact reason and it's because `forceEnabled()` occasionally gets changed without the instance stage being called again. Therefore, I decided to cache the parallelism flags at the instance stage (since that will never change), but I cache the non/parallel forces at Dynamics because they use forceEnabled.

>>forceEnabled at Stage::Topology
>>(1,1)
>>forceEnabled at Stage::Dynamics
>>(1,1)
>>forceEnabled at Stage::Dynamics
>>(0,1) <- If I only cached the forces at Instance, it would give me the wrong answer